enumerable/concat_test.go

```diff
@@ -0,0 +1,130 @@
+package enumerable
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestConcatYieldsNothingGivenEmpty(t *testing.T) {
```

Comment:
suggestion: may try the agreed upon test name structure: `<category>_<condition>_<result>`?

This example would be `TestConcat_IfEmpty_YieldNothing`

Reply:
I had already added tests before that conversation took place, and I'd prefer not to mix and match within a PR, and I'd also rather not rename everything.  The agreement was more of a 'lets try this for a bit' agreement, not a 'lets convert the entire codebase right now' to this agreement.